client/src/features/createPost/components/CreatePostModal.jsx

```diff
@@ -1,28 +1,71 @@
 import PropTypes from 'prop-types';
-import React, { useRef } from 'react';
+import React, { useRef, useState } from 'react';
+import { useMutation } from 'react-query';
 import useForm from '../../../hooks/useForm';
 import useOnClickOutside from '../../../hooks/useOnClickOutside';
+import { createPost, uploadImage } from '../../../utils/api';
 import capitalize from '../../../utils/helpers';
 import CreatePostSelect from './CreatePostSelect';
+import ImageDropzone from './ImageDropzone';
+import ImagePreview from './ImagePreview';
 
 const CreatePostModal = ({ isOpen, setIsOpen, firstName, children }) => {
-  const { formData, handleInputChange } = useForm({
+  const [files, setFiles] = useState([]);
+  const [select, setSelect] = useState('select visibility');
+  const [showDropzone, setShowDropzone] = useState(false);
+  const { formData, handleInputChange, reset } = useForm({
     text: '',
   });
+
+  const addPost = useMutation(createPost, {
+    onSuccess: () => {
+      setFiles([]);
+      reset();
+      setSelect('select visibility');
+      setShowDropzone(false);
+      setIsOpen(false);
+    },
+  });
+
+  const upload = useMutation(uploadImage, {
+    onSuccess: (data) => {
+      const {
+        data: { imageUrl },
+      } = data;
+      const newFormData = { ...formData, visibility: select, imageUrl };
+      addPost.mutate(newFormData);
+    },
+  });
+
+  const handleSubmit = () => {
+    if (!files?.length) {
+      addPost.mutate({ ...formData, visibility: select });
+    } else {
+      upload.mutate(files);
+    }
+  };
+
   const ref = useRef();
   useOnClickOutside(ref, () => {
     setIsOpen(false);
```

Comment:
I think here we should also clear fromData and files, otherwise, when we open the modal again we get the old existing data.
This should happen in both cases: if we click outside or if we click the close button.

Reply:
I checked the modal from facebook and they never clear the data unless u manually clear it or press the Post button. Do u still want the data to be cleared?

Reply:
we are not facebook haha. 😁 
What do u think then about asking the user if he wants to keep changes or remove them?

Reply:
We can keep this for later release also but we can keep it in mind if u agree